README.md

```diff
@@ -112,8 +112,8 @@ impl PoseidonLeaf<MemStore> for DataLeaf {
     }
 
     // Method used to set the position on the tree after the `PoseidonTree::push` call
-    fn tree_pos_mut(&mut self) -> &mut u64 {
-        &mut self.pos
+    fn tree_pos_set(&mut self, pos: u64) {
```

Comment:
Is the `tree` prefix mandatory? Couldn't the trait just needs to implement `pos()` (the getter) and `set_pos()` (the setter) as phoenix's note does?

If you think it's not clear enough is ok, I'll leave this up to you, but the naming in that case should be `set_tree_pos`.

Reply:
We have (imo) a limitation in Rust that will not grant namespacing for trait implementations. So if we have a very generic name for a trait that should be very generic, as is the `PoseidonLeaf`, then if the implementor needs to implement a method with the same name, the references to the method will be very ugly such as `<SomeType as PoseidonLeaf>::set_pos(&mut leaf, pos)`

The `tree_` prefix is a sort of workaround for the lack of a namespace, but I'm honestly not happy with this solution.

I could rename them both to `pos` and `set_pos`; what do you think?

Reply:
It's basically the issue we had with the `read` impl but I'm fine with that because it's only happened because the macro. We can ensure we're using the proper trait just doing `PoseidonLeaf::set_pos(&mut leaf, pos)` whenever arise the issue – it means, both of the trait are in the same scope.

I'm fine then with just use `pos` and `set_pos`, then.

Reply:
Done